custom_components/noonlight/switch.py

```diff
@@ -73,15 +73,65 @@ async def async_turn_on(self, **kwargs):
         #   whether medical, fire, or police should be notified
         if self._alarm is None:
             try:
-                self._alarm = await self.noonlight.client.create_alarm(
-                    body={
+
+                # dispatch is update to V2 noonlight API, added change here to use address
+                # for location reported to noonlight and also include desired service and
+                # special interaction instructions
+                if "dispatch" in self.noonlight.config.get('api_endpoint'):
+
+                    # if you have a dev_token you can set it in the configuration so that you can
+                    # view the messages you're generating in the noonlight developer web site
+                    if 'dev_token' in self.noonlight.config.keys():
+                        self.noonlight.client.set_token( token = self.noonlight.config.get('dev_token') )
+                        _LOGGER.debug('noonlight client token set. '
+                               'token: %s', self.noonlight.config.get('dev_token'))
+
+                    # location and primary contact are required items in V2
+                    body_param={
+                        'location':{
+                            'address': {
+                                'line1': self.noonlight.config.get('line1'),
+                                'line2': self.noonlight.config.get('line2') or "N/A",
+                                'city': self.noonlight.config.get('city'),
+                                'state': self.noonlight.config.get('state'),
+                                'zip': self.noonlight.config.get('zip')
+                            }
+                        },
+                        'name': self.noonlight.config.get('name'),
+                        'phone': self.noonlight.config.get('phone'),
+                        'pin': self.noonlight.config.get('pin')
+                    }
+
+                    #try:
```

Comment:
This commented out code should be removed

Reply:
I accepted those that gave me that options and then made the requested changes to my copy of the code.  I'm old school and have only done a few pull request via github.  What's the best way to have these updates included?  Should I delete the current pull request and issue another, or generate a second pull request?  Thanks

Reply:
> I accepted those that gave me that options and then made the requested changes to my copy of the code. I'm old school and have only done a few pull request via github. What's the best way to have these updates included? Should I delete the current pull request and issue another, or generate a second pull request? Thanks

No need to create another pull request - this one is updated to include all the commits you added.

Reply:
Isn't that nice, looking below it shows all of the changes.  I guess this means they are automatically included in the pull request.